src/main/java/frc/robot/subsystems/GameObjectManager.java

```diff
@@ -165,13 +163,36 @@ public int bestPlace() {
 
     @Override
     public void periodic() {
-        Collection<Detection> detections = m_coprocessor.getAllDetections();
-        for (Detection d : detections) {
-            Pose2d pos = new Pose2d(d.getPosition().x, d.getPosition().y, new Rotation2d(0.));
-            Pose2d globalpos = pos.transformBy(new Transform2d(m_coprocessor.getTagGlobalPose(), new Pose2d()));
-            addGameObject(d.getName(), globalpos.getX(), globalpos.getY(), d.getPosition().z, 0);
+        if (m_coprocessor.isTagGlobalPoseActive()) {
+            Collection<Detection> detections = m_coprocessor.getAllDetections();
+            for (Detection d : detections) {
+                Pose2d pos = new Pose2d(d.getPosition().x, d.getPosition().y, new Rotation2d(0.));
+                Pose2d globalpos = pos.transformBy(new Transform2d(m_coprocessor.getTagGlobalPose(), new Pose2d()));
+                addGameObject(d.getName(), globalpos.getX(), globalpos.getY(), d.getPosition().z, 0);
+            }
+            removeInactiveGameObjects();
+            fillGridZones();
+
+            int closestColumnIndex = 0;
+            double distance = 1000; 
+            for (int i = 0; i < 9; i++) {
+                if (gridZones.get(i).getX()-m_coprocessor.getTagGlobalPoseInches().getX() < distance) {
```

Comment:
I think you want to wrap this in Math.abs()

Reply:
also, I would store this value in a variable since you do the same computation a few lines down